paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -439,14 +441,41 @@ protected void alterTableImpl(Identifier identifier, List<SchemaChange> changes)
             // sync to hive hms
             Table table = client.getTable(identifier.getDatabaseName(), identifier.getObjectName());
             updateHmsTable(table, identifier, schema);
-            client.alter_table(
-                    identifier.getDatabaseName(), identifier.getObjectName(), table, true);
+            alterTableByEnvContext(identifier.getDatabaseName(), identifier.getObjectName(), table);
         } catch (Exception te) {
             schemaManager.deleteSchema(schema.id());
             throw new RuntimeException(te);
         }
     }
 
+    /**
+     * Get EnvironmentContext with cascade param.
+     *
+     * @return EnvironmentContext
+     */
+    private EnvironmentContext getEnvironmentContext() {
+        EnvironmentContext environmentContext = new EnvironmentContext();
+        environmentContext.putToProperties(
+                StatsSetupConst.CASCADE, StatsSetupConst.TRUE);
+        return environmentContext;
+    }
+
+    private void alterTableByEnvContext(String databaseName, String objectName, Table table)
+            throws TException {
+        boolean skipTableStatsProperty =
+                hiveConf.getBoolean(
+                        HiveCatalogOptions.TABLE_STATS_SKIP.key(),
+                        HiveCatalogOptions.TABLE_STATS_SKIP.defaultValue());
+        EnvironmentContext environmentContext = getEnvironmentContext();
+        if (skipTableStatsProperty) {
+            // Avoid Hive metastore read SD location path data.
+            environmentContext.putToProperties(
+                    StatsSetupConst.DO_NOT_UPDATE_STATS, StatsSetupConst.TRUE);
+        }
+        client.alter_table_with_environmentContext(
```

Comment:
Because we need to support multiple versions of hive, can the alter here be done using reflection and compatible with multiple versions?

You can take a look to iceberg: `org.apache.iceberg.hive.MetastoreUtil`.